### ISSUE TITLE
Install on Ubuntu 18.04, encoder encodes and no output videos appear to be generated.

### ISSUE THREAD

**Issue by walt93** [WWBN/AVideo#4585]
**Describe the bug**
Add files to encode queue. Watch files encode. Come back later. Find no media visible in the UI. Can't find encoded media on disk. Have a bunch of errors.

**To Reproduce**
1. Mostly followed these instructions https://www.linuxbabe.com/ubuntu/install-youphptube-ubuntu-18-04-server

**Expected behavior**
After encoding a video, I expect it to be visible in the UI, able to edit its metadata, play it, watch view counts increase, ...

Ubuntu 18.0.4, browser seems not to matter but Safari Firefox and Brave, current version downloaded ≈1 week ago so feb 14-ish.

**Error Logs**
[21-Feb-2021 11:30:21 UTC] PHP Notice:  Trying to get property 'video_id' of non-object in /var/www/AVideo/upload/objects/Encoder.php on line 1198
[21-Feb-2021 11:30:21 UTC] Encoder:sendFileChunk(/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__480.mp4, 0, mp4, object, 480, 3)
[21-Feb-2021 11:30:21 UTC] Encoder::sendFileChunk file=/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__480.mp4
[21-Feb-2021 11:30:21 UTC] AVideo-Streamer countCURLOPT_READFUNCTION = (0) chunk answer <html>
<head><title>405 Not Allowed</title></head>
<body bgcolor="white">
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:21 UTC] cURL error, trying again (4) (0):
 No error
 /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__480.mp4
 (https://tube.conspyre.tv/aVideoEncoderChunk.json)
[21-Feb-2021 11:30:21 UTC] {"error":true,"file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__480.mp4","filesize":426831964,"response":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoderChunk.json","response_raw":"<html>\r\n<head><title>405 Not Allowed<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>405 Not Allowed<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","msg":"cURL error (0):\n No error \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__480.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoderChunk.json)"}
[21-Feb-2021 11:30:21 UTC] Encoder::sendFile videos_id=0, format=mp4
[21-Feb-2021 11:30:22 UTC] Duration found: 1:59:58
[21-Feb-2021 11:30:22 UTC] Encoder::sendFile sending file to https://tube.conspyre.tv/aVideoEncoder.json
[21-Feb-2021 11:30:22 UTC] Encoder::sendFile reading file from /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__480.mp4
[21-Feb-2021 11:30:22 UTC] AVideo-Streamer answer <html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:22 UTC] {"error":true,"format":"mp4","file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__480.mp4","resolution":"480","videoDownloadedLink":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoder.json","postFields":15,"videoFileSize":"407.06MB","response_raw":"<html>\r\n<head><title>404 Not Found<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","response":null,"msg":"cURL error (1):\n Unsupported protocol \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__480.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoder.json)\n 4 "}
[21-Feb-2021 11:30:22 UTC] PHP Notice:  Trying to get property 'video_id' of non-object in /var/www/AVideo/upload/objects/Encoder.php on line 1009
[21-Feb-2021 11:30:22 UTC] Encoder::send() multiResolutionOrder sendAll resolution(720) (/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4)
[21-Feb-2021 11:30:22 UTC] Encoder::sendFileToDownload videos_id=0, format=mp4
[21-Feb-2021 11:30:22 UTC] Duration found: 1:59:58
[21-Feb-2021 11:30:22 UTC] Encoder::sendFileToDownload sending file to https://tube.conspyre.tv/aVideoEncoder.json
[21-Feb-2021 11:30:22 UTC] Encoder::sendFileToDownload reading file from /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:22 UTC] AVideo-Streamer answer <html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:22 UTC] {"error":true,"format":"mp4","file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4","resolution":"720","videoDownloadedLink":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoder.json","postFields":12,"videoFileSize":"726.76MB","response_raw":"<html>\r\n<head><title>404 Not Found<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","response":null,"msg":"cURL error (1):\n Unsupported protocol \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoder.json)\n "}
[21-Feb-2021 11:30:22 UTC] PHP Notice:  Trying to get property 'video_id' of non-object in /var/www/AVideo/upload/objects/Encoder.php on line 1198
[21-Feb-2021 11:30:22 UTC] Encoder:sendFileChunk(/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4, 0, mp4, object, 720, 0)
[21-Feb-2021 11:30:22 UTC] Encoder::sendFileChunk file=/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:22 UTC] AVideo-Streamer countCURLOPT_READFUNCTION = (0) chunk answer <html>
<head><title>405 Not Allowed</title></head>
<body bgcolor="white">
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:22 UTC] cURL error, trying again (1) (0):
 No error
 /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
 (https://tube.conspyre.tv/aVideoEncoderChunk.json)
[21-Feb-2021 11:30:23 UTC] Encoder::sendFileToDownload videos_id=0, format=mp4
[21-Feb-2021 11:30:23 UTC] Duration found: 1:59:58
[21-Feb-2021 11:30:23 UTC] Encoder::sendFileToDownload sending file to https://tube.conspyre.tv/aVideoEncoder.json
[21-Feb-2021 11:30:23 UTC] Encoder::sendFileToDownload reading file from /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:23 UTC] AVideo-Streamer answer <html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:23 UTC] {"error":true,"format":"mp4","file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4","resolution":"720","videoDownloadedLink":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoder.json","postFields":12,"videoFileSize":"726.76MB","response_raw":"<html>\r\n<head><title>404 Not Found<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","response":null,"msg":"cURL error (1):\n Unsupported protocol \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoder.json)\n "}
[21-Feb-2021 11:30:23 UTC] PHP Notice:  Trying to get property 'video_id' of non-object in /var/www/AVideo/upload/objects/Encoder.php on line 1198
[21-Feb-2021 11:30:23 UTC] Encoder:sendFileChunk(/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4, 0, mp4, object, 720, 1)
[21-Feb-2021 11:30:23 UTC] Encoder::sendFileChunk file=/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:23 UTC] AVideo-Streamer countCURLOPT_READFUNCTION = (0) chunk answer <html>
<head><title>405 Not Allowed</title></head>
<body bgcolor="white">
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:23 UTC] cURL error, trying again (2) (0):
 No error
 /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
 (https://tube.conspyre.tv/aVideoEncoderChunk.json)
[21-Feb-2021 11:30:25 UTC] Encoder::sendFileToDownload videos_id=0, format=mp4
[21-Feb-2021 11:30:25 UTC] Duration found: 1:59:58
[21-Feb-2021 11:30:25 UTC] Encoder::sendFileToDownload sending file to https://tube.conspyre.tv/aVideoEncoder.json
[21-Feb-2021 11:30:25 UTC] Encoder::sendFileToDownload reading file from /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:25 UTC] AVideo-Streamer answer <html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:25 UTC] {"error":true,"format":"mp4","file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4","resolution":"720","videoDownloadedLink":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoder.json","postFields":12,"videoFileSize":"726.76MB","response_raw":"<html>\r\n<head><title>404 Not Found<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","response":null,"msg":"cURL error (1):\n Unsupported protocol \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoder.json)\n "}
[21-Feb-2021 11:30:25 UTC] PHP Notice:  Trying to get property 'video_id' of non-object in /var/www/AVideo/upload/objects/Encoder.php on line 1198
[21-Feb-2021 11:30:25 UTC] Encoder:sendFileChunk(/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4, 0, mp4, object, 720, 2)
[21-Feb-2021 11:30:25 UTC] Encoder::sendFileChunk file=/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:25 UTC] AVideo-Streamer countCURLOPT_READFUNCTION = (0) chunk answer <html>
<head><title>405 Not Allowed</title></head>
<body bgcolor="white">
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:25 UTC] cURL error, trying again (3) (0):
 No error
 /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
 (https://tube.conspyre.tv/aVideoEncoderChunk.json)
[21-Feb-2021 11:30:28 UTC] Encoder::sendFileToDownload videos_id=0, format=mp4
[21-Feb-2021 11:30:28 UTC] Duration found: 1:59:58
[21-Feb-2021 11:30:28 UTC] Encoder::sendFileToDownload sending file to https://tube.conspyre.tv/aVideoEncoder.json
[21-Feb-2021 11:30:28 UTC] Encoder::sendFileToDownload reading file from /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:28 UTC] AVideo-Streamer answer <html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:28 UTC] {"error":true,"format":"mp4","file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4","resolution":"720","videoDownloadedLink":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoder.json","postFields":12,"videoFileSize":"726.76MB","response_raw":"<html>\r\n<head><title>404 Not Found<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","response":null,"msg":"cURL error (1):\n Unsupported protocol \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoder.json)\n "}
[21-Feb-2021 11:30:28 UTC] PHP Notice:  Trying to get property 'video_id' of non-object in /var/www/AVideo/upload/objects/Encoder.php on line 1198
[21-Feb-2021 11:30:28 UTC] Encoder:sendFileChunk(/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4, 0, mp4, object, 720, 3)
[21-Feb-2021 11:30:28 UTC] Encoder::sendFileChunk file=/var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:28 UTC] AVideo-Streamer countCURLOPT_READFUNCTION = (0) chunk answer <html>
<head><title>405 Not Allowed</title></head>
<body bgcolor="white">
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:28 UTC] cURL error, trying again (4) (0):
 No error
 /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
 (https://tube.conspyre.tv/aVideoEncoderChunk.json)
[21-Feb-2021 11:30:28 UTC] {"error":true,"file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4","filesize":762058827,"response":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoderChunk.json","response_raw":"<html>\r\n<head><title>405 Not Allowed<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>405 Not Allowed<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","msg":"cURL error (0):\n No error \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoderChunk.json)"}
[21-Feb-2021 11:30:28 UTC] Encoder::sendFile videos_id=0, format=mp4
[21-Feb-2021 11:30:29 UTC] Duration found: 1:59:58
[21-Feb-2021 11:30:29 UTC] Encoder::sendFile sending file to https://tube.conspyre.tv/aVideoEncoder.json
[21-Feb-2021 11:30:29 UTC] Encoder::sendFile reading file from /var/www/AVideo/upload/videos/avideoTmpFile_5_streamers_id_2__720.mp4
[21-Feb-2021 11:30:29 UTC] AVideo-Streamer answer <html>
<head><title>404 Not Found</title></head>
<body bgcolor="white">
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.14.0 (Ubuntu)</center>
</body>
</html>

[21-Feb-2021 11:30:29 UTC] {"error":true,"format":"mp4","file":"\/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4","resolution":"720","videoDownloadedLink":null,"target":"https:\/\/tube.conspyre.tv\/aVideoEncoder.json","postFields":15,"videoFileSize":"726.76MB","response_raw":"<html>\r\n<head><title>404 Not Found<\/title><\/head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found<\/h1><\/center>\r\n<hr><center>nginx\/1.14.0 (Ubuntu)<\/center>\r\n<\/body>\r\n<\/html>\r\n","response":null,"msg":"cURL error (1):\n Unsupported protocol \n \/var\/www\/AVideo\/upload\/videos\/avideoTmpFile_5_streamers_id_2__720.mp4 \n (https:\/\/tube.conspyre.tv\/aVideoEncoder.json)\n 4 "}
[21-Feb-2021 11:30:29 UTC] PHP Notice:  Trying to get property 'video_id' of non-object in /var/www/AVideo/upload/objects/Encoder.php on line 1009
[21-Feb-2021 11:30:29 UTC] Encoder::save id=(5) title=(Absolue Proof-VKaSX9SKR1A bestvideo+bestaudio)
[21-Feb-2021 11:30:29 UTC] getReturn_vars {"videos_id":null}
[21-Feb-2021 11:30:29 UTC] {}
[21-Feb-2021 11:30:29 UTC] Encoder::save id=(5) title=(Absolue Proof-VKaSX9SKR1A bestvideo+bestaudio)

**Comment by walt93** [comment]
UPDATE: RTFM when it says "NGINX IS NOT SUPPORTED".

TL;dr I was using Nginx and my AVideo site sucked. I switched to Apache, sorted out its issues with my other app, and AVideo rucking fox!
